opencog/nlp/scm/processing-utils.scm

```diff
@@ -127,3 +133,160 @@
+			(if (not a-match)
+				(list a-string)
+				(append (list (match:prefix a-match))
+					(split-string a-pattern (match:suffix a-match)))
+			)
+		)
+	)
+
+	; ---------------------------------------------------------------------
+	; returns the name of a ParseNode if the string has a ParseNode entry.
+	; returns "sentence@dbce9f0d-8b8a-4ea7-a8c8-d69abf05f810_parse_1" from the string
+	; "84590fb2e46c (ParseNode \"sentence@dbce9f0d-8b8a-4ea7-a8c8-d69abf05f810_parse_1\""
+	(define (parse-str a-string)
+		(substring (match:substring
+			(string-match "ParseNode \"sentence@[[:alnum:]_-]*" a-string)) 11)
+	)
+
+	; ---------------------------------------------------------------------
+	; Evaluate the string and returns a list with SetLink of the initial r2l rule application.
+	; (ReferenceLink 
+	;   (InterpretationNode "sentence@1d220-c7ace2f_parse_2_interpretation_$X")
+	;   (SetLink
+	;       different links that are a result of r2l rule-functions being applied
+	;   )
+	; )
+	;
+	; A ReferenceLink is used instead of InterpretationLink so as to differentiate
+	; the final word-sense-disambiguated , anaphore and cataphor resolved version from
+	; the initial r2l output. The final version will have structure that is detailed
+	; @ http://wiki.opencog.org/w/Linguistic_Interpretation.
+	; Having the output from this function should help during garbage-removal from the
+	; atomspace as well as the generation of the final set of interpretations for the
+	; the sentence.
+	(define (set-link a-string)
+		; z-list is a list containing a set of lists, with firts elements of the
+		; sub-list being a relex-opencog-output string and the second element being
+		; a string of the relex-to-logic function calls that is to be applied on the
+		; relex-opencog-output in the atomspace. The last sub-list is just the AnchorNode.
+		(define z-list (map (lambda (x) (split-string pattern2 x)) 
+		                                (split-string pattern1 a-string)))
+
+		; helper function that prune away atoms that no longer exists from subsequent
+		; rules, or atoms that are wrapped inside another link
+		(define (pruner x)
+			(define deref-x (cog-atom (cog-handle x)))
+			; if 'deref-x' is #<Invalid handle>, than both cog-node?
+			; and cog-link? will return false
+			(if (and (or (cog-node? deref-x) (cog-link? deref-x))
+			         (null? (cog-incoming-set x)))
+				x
+				#f
+			)
+		)
+
+		; Given any one of the sub-lists from the z-lists it will evaluate the strings
+		; elements from left to right resulting in the creation of the Atoms of the 
+		; relex-to-logic pipeline.
+		(define (eval-list a-list)
+			(if (= 1 (length a-list))
+				(begin (eval-string (list-ref a-list 0)) '())
+				(begin
+					(eval-string (list-ref a-list 0))
+					(let* ((parse-name (parse-str (list-ref a-list 0)))
+					       (parse-node (ParseNode parse-name)))
+;
+; XXX not really necessary at this point since we will only need the LG
```

Comment:
Humm... it was on purpose because it's not really a commenting the code but instead is commenting about the commenting out of an entire block of code.